### PR TITLE
UI create calendar component

### LINKED ISSUE #56: UI: Create calendar component
Create a calendar component 

* Ideally generic/modular to be able to use for both the availabilities and flights views
* *link to figma?*

### PR DESCRIPTION
**Pull request by hacknstack**
Pull Request Description
------------------------

### Overview
Close #56 

This introduces a set of functions and composables for managing and displaying availability status within a calendar view. It enhances the user interface by providing visual cues about the availability of time slots.
It works like this, the tiles are first shown based on the getAvailabilityStatus and everytime you click, it updates them with the nextAvailabilityStatus function to change the tiles color.

### Changes Made

*   Added dummy functions `getAvailabilityStatus` and `nextAvailabilityStatus` to determine and update the availability status for a given date and time slot.
*   Introduced a composable function `showTile` to display colored tiles indicating the availability status.
*   Implemented `showCalendarAvailabilities` to display a calendar view with availability tiles using Jetpack Compose.
*   Added `Calendar` and `WeekView` composables for rendering a calendar layout with customizable tiles for each day.
*   Included a utility function `getStartOfWeek` to calculate the start date of the week based on a given date.

### Android render:

IMG_1293

### Additional infos:
The part for the models should be review somewhere else.

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/MainActivity.kt
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/MainActivity.kt:42]: You shouldn't modify the `MainActivity` file (it's great for testing but you shouldn't commit it)

**Review by Plouc314 — APPROVED** [review]
LGTM, good base of Calendar.
You just have to wait on the #75 PR before you merge (because you're branch is based on it) and revert the changes on `MainActivity`.

File: app/src/main/java/ch/epfl/skysync/MainActivity.kt
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/MainActivity.kt:42]: I agree it was a git mistake

**Pull request closed without merging**
